pkg/sources/filesystem/filesystem.go

```diff
@@ -270,7 +270,11 @@ func (s *Source) Enumerate(ctx context.Context, reporter sources.UnitReporter) e
 
 // ChunkUnit implements SourceUnitChunker interface.
 func (s *Source) ChunkUnit(ctx context.Context, unit sources.SourceUnit, reporter sources.ChunkReporter) error {
-	path := unit.SourceUnitID()
+	commonUnit, err := sources.IntoCommonUnit(unit)
+	if err != nil {
+		return err
+	}
+	path := commonUnit.ID
```

Comment:
Having to do this whole song and dance in every source that doesn't implement its own unit type just to get an `ID` field out of the unit surprises me, given the existence of a `SourceUnitID()` method on the interface. It looks like the friction is caused by `SourceUnitID()` and `CommonSourceUnit.ID` being two different "source unit IDs" that serve two different purposes and are consumed by two different parts of the code. Is _that_ something that it makes sense to change? Perhaps we could rename one of them in order to sensibly provide an "actual" ID method on the interface, and thus avoid these type assertions in most cases...but now I'm wondering why we need a `Kind` on `CommonSourceUnit` at all. There's only ever one kind of common source unit (the, uh, common one). So what's the field for?

Reply:
Yeah, I had a similar thought.. `ID` is no longer the `ID`, though for the common unit it probably should be?

I was intending for `Kind` to be a hint for the UI about what the unit is.

Reply:
Can you extend the `SourceUnit` interface with a first-class method exposing kind and stop bundling the kind into `SourceUnitID()`?